src/parser/mod.rs

```diff
@@ -10307,11 +10307,67 @@ impl<'a> Parser<'a> {
                 snapshot: None,
                 session: false,
             })
+        } else if self.dialect.supports_set_stmt_without_operator() {
+            self.prev_token();
+            self.parse_set_session_params()
         } else {
             self.expected("equals sign or TO", self.peek_token())
         }
     }
 
+    pub fn parse_set_session_params(&mut self) -> Result<Statement, ParserError> {
+        let names = self.parse_comma_separated(Parser::parse_set_session_param_name)?;
+
+        let last_name = if let Some(last_name) = names.last() {
+            last_name.to_uppercase()
+        } else {
+            return self.expected("Session param name", self.peek_token());
+        };
+
+        let identity_insert_obj = if last_name == "IDENTITY_INSERT" {
+            Some(self.parse_object_name(false)?)
+        } else {
+            None
+        };
+
+        let offsets_keywords = if last_name == "OFFSETS" {
+            Some(self.parse_comma_separated(|parser| {
+                let next_token = parser.next_token();
+                match &next_token.token {
+                    Token::Word(w) => Ok(w.to_string()),
+                    _ => parser.expected("SQL keyword", next_token),
+                }
+            })?)
+        } else {
+            None
+        };
+
+        let value = self.parse_expr()?.to_string();
+        Ok(Statement::SetSessionParam {
+            names,
+            identity_insert_obj,
+            offsets_keywords,
+            value,
+        })
+    }
+
+    pub fn parse_set_session_param_name(&mut self) -> Result<String, ParserError> {
+        if self.parse_keywords(&[Keyword::STATISTICS, Keyword::IO]) {
```

Comment:
Given these are hard coded it seems like it would be cleaner to eventually return an enum for what setting was set rather than arbitrary strings that the using library needs to parse / interpret

However, I also think we could do that as a follow on PR